classes/System.php

```diff
@@ -699,6 +700,12 @@ public function checkForActiveEvent(): void {
         if($this->testNotifications['event'] && is_null($this->event) && $this->isDevEnvironment()) {
             $this->event = new DoubleExpEvent($current_datetime->modify("+2 weeks"));
         }
+
+        // If no other event active on weekend, activate bonus XP
+        if (!isset($this->event) && (date('w') == 0 || date('w') == 6)) {
```

Comment:
```suggestion
        if (!isset($this->event) && (System::currentDayOfWeek() == 0 || System::currentDayOfWeek() == 6)) {
```

Can you add a helper for this (we have the others for currentDay/CurrentMonth/etc already):

```
    /**
     * @return int day of the week: 0 (Sunday) - 6 (Saturday)
     */
    public static function currentDayOfWeek(): int {
        return (int) date('w');
    }
```

Reply:
Sure!